src/casync.c

```diff
@@ -200,6 +201,21 @@ CaSync *ca_sync_new_decode(void) {
         return s;
 }
 
+int ca_sync_set_exclude_from(CaSync *s, const char *path) {
+
+        if (!s)
+                return -EINVAL;
+
+        if (s->direction != CA_SYNC_ENCODE)
+                return -EINVAL;
+
+        s->exclude_from = strdup(path);
```

Comment:
If `s->exclude_from` was previously set, it needs to be freed before reassigning.
A corresponding free at the end is also needed. Prolly where `archive_path` is being freed.